### PR TITLE
Xenbase load

### PR DESCRIPTION
**Pull request by kevinschaper**

### CODE REVIEW

File: examples/xenbase/gene-literature.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        gene = Gene()
+        gene.id = gene_id
+
+        entities.append(gene)
+
+        association = NamedThingToInformationContentEntityAssociation()
+        association.subject = gene.id
+        association.predicate = "IAO:0000142"  # Mentions
+        association.object = publication.id
+        association.relation = "RO:0001019"  # Contains todo: confirm this!
```
Comment (kevinschaper) [inline examples/xenbase/gene-literature.py:41]: What do you think about the relation type here @kshefchek @sierra-moxon?

File: koza/model/source.py [hunk 2]
```diff
@@ -98,4 +100,5 @@ def _get_row(self):
                 row = next(self._reader)
         else:
             row = next(self._reader)
+        self.last_row = row
```
Comment (kshefchek) [inline koza/model/source.py:103]: curious why this is needed? maybe some comments would help here

**Review by kshefchek — APPROVED** [review]

File: koza/model/source.py (hunk 2 above)
Comment (kevinschaper) [reply koza/model/source.py:103]: Oh, good call, that's in there to print the row that won't validate alongside the validation exception, but that's definitely not self explanatory from the code.

File: koza/io/reader/csv_reader.py [hunk 3]
```diff
@@ -174,13 +183,22 @@ def __next__(self) -> Dict[str, Any]:
 
         # if we've made it here we can convert a row to a dict
         field_map = dict(zip(self.fieldnames, row))
+
+        # to avoid KeyErrors / error handling in source scripts, add missing fields to the map
+        for fieldname in self.fieldnames:
+            if fieldname not in field_map.keys():
+                field_map[fieldname] = None
```
[the remark this answers is not in the document]
Comment (kevinschaper) [reply koza/io/reader/csv_reader.py:190]: I could probably revert that, I agree about only allowing it for end of row missing columns, but also it was in there for the Xenbase gene file that we ended up not using.

**Pull request merged**